credential/signature.go

```diff
@@ -0,0 +1,110 @@
+package credential
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	"github.com/TBD54566975/ssi-sdk/crypto"
+	"github.com/TBD54566975/ssi-sdk/did"
+	"github.com/pkg/errors"
+)
+
+// VerifyCredentialSignature verifies the signature of a credential of any type
+// TODO(gabe) support other types of credentials https://github.com/TBD54566975/ssi-sdk/issues/352
+func VerifyCredentialSignature(ctx context.Context, genericCred any, resolver did.Resolver) (bool, error) {
+	if genericCred == nil {
+		return false, errors.New("credential cannot be empty")
+	}
+	if resolver == nil {
+		return false, errors.New("resolver cannot be empty")
+	}
+	switch genericCred.(type) {
+	case *VerifiableCredential:
+		return VerifyCredentialSignature(ctx, *genericCred.(*VerifiableCredential), resolver)
```

Comment:
Are we sure we won't run into an infinite loop problem here or at the other 
return VerifyCredentialSignature spots?

Reply:
sure...maybe not. I have each case covered in a test that should be "good enough"